### PR TITLE
Convert to class to reduce argument passing

### PR DESCRIPTION
**Pull request by mads-hartmann**
## Description

This promotes `installMonitoringSatellite` to be a class to reduce the need for passing the same arguments around to the functions.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/9059

## How to test

- [x] in core-dev with observability enabled `werft run github -f -a with-helm=true -a with-observability=true` - [job](https://werft.gitpod-dev.com/job/gitpod-build-mads-classify-monitoring-satellite.3) - was able to port-forward to Prometheus and Grafana.
- [x] with-vm ` werft run github -f -a with-vm=true` - [job](https://werft.gitpod-dev.com/job/gitpod-build-mads-classify-monitoring-satellite.4) - was able to port-forward to Prometheus and Grafana but had to use the following commands instead of our own script. I suspect there's a bug in our proxy. Will open a follow up issue.
  ```
  kubectl --context=k3s-preview-environment port-forward svc/prometheus-k8s "9090:9090"
  kubectl --context=k3s-preview-environment port-forward svc/grafana "3000:3000"
  ```

## Release Notes

```release-note
NONE
```

## Documentation

N/A

### CODE REVIEW

File: .werft/observability/monitoring-satellite.ts
Comment (ArthurSens) [inline .werft/observability/monitoring-satellite.ts:139]: back when I implemented this I had no idea what was this `async` and was just adding because it was being used everywhere.

Does it make sense in this class? Almost function is using async, although we're not using Promises anywhere. Would it make sense to just remove it?

File: .werft/jobs/build/deploy-to-preview-environment.ts
Comment (ArthurSens) [inline .werft/jobs/build/deploy-to-preview-environment.ts:774]: same thing here, we declare an `async` function only to use `await` when calling it. Would it make sense to just remove it?

**Review by ArthurSens — APPROVED** [review]
Noice! I'm really liking this pattern for using those "installer" classes. 

I have a few questions but I don't know if they make sense 😅, so I'm approving with a hold. Please decide if we still need to make changes here, leave them another PR or not do them at all

/hold

**Comment by mads-hartmann** [comment]
@ArthurSens Thanks for the pointer! I removed the async keyword from the methods and also the async option for the exec invocations - in effect making observability installation blocking. Previously we were blocking using `await` in core-dev but not in Harvester. As we generally want observability ready before we install Gitpod to avoid errors about Gitpod not being able to send telemetry I think it makes sense to always have the function be blocking. But I will test the code tomorrow and do another pass to see if it makes sense ☺️

**Comment by ArthurSens** [comment]
I believe this is good to go? Can I just remove the hold? 🙂

**Comment by mads-hartmann** [comment]
@ArthurSens When testing earlier it looked like it had broken tracing but I didn't have time to debug further. I will move this back to draft ☺️

File: .werft/jobs/build/deploy-to-preview-environment.ts
Comment (mads-hartmann) [reply .werft/jobs/build/deploy-to-preview-environment.ts:797]: I'm keeping this async as it's my understand that we want to block in core-dev (when observability is enabled) but not block when installing it in VM-based preview environments ☺️

**Comment by mads-hartmann** [comment]
Okay @ArthurSens this is finally ready for another review. In the future I think we want to separate the installation of monitoring satellite from the waiting for it to be ready. That way we'd avoid the problem of having a slice potentially spanning multiple phases, but I didn't want to make that change now as I've already spent too long on this 😅

This should PR should also fix the problem Chris has been experiencing with failed monitoring satellite deployments breaking the build as explained [here](https://gitpod.slack.com/archives/C01KGM9EBD4/p1651144271318639?thread_ts=1651061591.320309&cid=C01KGM9EBD4).

The last job has some missing spans, but I think that's more likely to to be due to burst protection as you also called out [here](https://gitpod.slack.com/archives/C01KGM9EBD4/p1651053173542839) - that also explains why I saw issue sometime yesterday but not always.

**Comment by mads-hartmann** [comment]
@ArthurSens lol sorry, I have to get used to commits not dismissing previous reviews - I did want you to have a look at this before merging 😅  I don't want to revert it as I tested it pretty well, but if you have time please have a look and I can create a follow up PR with any changes you'd like to see

**Review by ArthurSens — COMMENTED** [review]
No worries about the merge, nothing really important to change :)

**Pull request merged**
